File: backend/Mission11.API/Data/Book.cs
using System.ComponentModel.DataAnnotations;

namespace Mission11.API.Data;

// Represents a book entity in the database
public class Book
{
    [Key]
    [Required]
    public int BookID { get; set; }
    
    [Required]
    public string Title { get; set; }
    
    [Required]
    public string  Author { get; set; }
    
    [Required]
    public string Publisher { get; set; }
    
    [Required]
    public string ISBN { get; set; }
    
    [Required]
    public string Classification { get; set; }
    
    [Required]
    public string Category { get; set; }
    
    [Required]
    public int PageCount { get; set; }
    
    [Required]
    public float Price { get; set; }
    
}

// what one record is going to look like

File: frontend/src/types/Book.ts
// Defines the structure of a bowler object
// This interface is used to enforce type-checking and ensure that
// each book has the specified properties with their corresponding data types.
//object that is coming in the json object
export interface Book {
  bookID: number;
  title: string;
  author: string;
  publisher: string;
  isbn: string;
  classification: string;
  category: string;
  pageCount: number;
  price: number;
}
